src/core/service.c

```diff
@@ -2579,11 +2579,12 @@ static void service_notify_cgroup_empty_event(Unit *u) {
                  * SIGCHLD for. */
 
         case SERVICE_START:
-        case SERVICE_START_POST:
                 if (s->type == SERVICE_NOTIFY)
                         /* No chance of getting a ready notification anymore */
                         service_enter_signal(s, SERVICE_FINAL_SIGTERM, SERVICE_FAILURE_PROTOCOL);
-                else if (s->pid_file_pathspec) {
+                else /* Fall through */
```

Comment:
uh, dropping into the next case as "else" statement looks strange. Quite franky, I am surprised this works at all... 

Can you rewrite this and make the first if branch end in a "break", so that we don't need the "else" here?